### PR TITLE
Use ImagePullPolicyIfNotPresent for HO

### PR DESCRIPTION
**Pull request by pcrentsil**
Do not always pull Hypershift images

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by enxebre** [comment]
/hold
Thanks @pcrentsil. [Please squash commits reasonably, use imperative mood and add a PR and commit desc elaborating "why" this change](https://github.com/openshift/hypershift/blob/main/docs/content/contribute/index.md).

File: contrib/admission-tracer/manifests/deployment.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ spec:
       - command:
         - /usr/bin/admission-tracer
         image: quay.io/sjenning/admission-tracer:latest
-        imagePullPolicy: Always
+        imagePullPolicy: IfNotPresent
         name: admission-tracer
```
Comment (hasueki) [inline contrib/admission-tracer/manifests/deployment.yaml:21]: @sjenning Do we want to keep this as `Always`? I see that the image references the `latest` tag.

Comment (alvaroaleman) [reply contrib/admission-tracer/manifests/deployment.yaml:21]: This is a dev debugging tool, I think Always makes more sense

File: control-plane-operator/controllers/hostedcontrolplane/kas/config.go [hunk 2]
```diff
@@ -191,7 +191,8 @@ func generateConfig(p KubeAPIServerConfigParams, version semver.Version) *kcpv1.
 	args.Set("service-account-lookup", "true")
 	args.Set("service-account-signing-key-file", cpath(kasVolumeServiceAccountKey().Name, pki.ServiceSignerPrivateKey))
 	args.Set("service-node-port-range", p.NodePortRange)
-	args.Set("shutdown-delay-duration", "10s")
+	args.Set("shutdown-delay-duration", "70s")
```
Comment (alvaroaleman) [inline control-plane-operator/controllers/hostedcontrolplane/kas/config.go:194]: I don't see how this is related to ImagePullPolicy. Please read our contributing guide: https://hypershift-docs.netlify.app/contribute/

/hold

**Comment by pcrentsil** [comment]
@alvaroaleman comments were updated, commits squashed please take a look. Thank you,

**Review by alvaroaleman — APPROVED** [review]
/retitle Use ImagePullPolicyIfNotPresent for HO
/hold cancel

**Pull request merged**